### ISSUE TITLE
Cannot infer type error is not reported

### ISSUE THREAD

**Issue by auduchinok** [intellij-rust/intellij-rust#6995]
```rust
fn main() {
    let mut guess = String::new();
    let guess: u32 = match guess.trim().parse() {};
}
```

```
error[E0284]: type annotations needed
 --> src/main.rs:4:41
  |
4 |     let guess: u32 = match guess.trim().parse() {};
  |                                         ^^^^^ cannot infer type for type parameter `F` declared on the associated function `parse`
  |
```

## Environment

* **IntelliJ Rust plugin version:** 0.3.143.3737-211
* **Rust toolchain version:** 1.50.0 aarch64-apple-darwin
* **IDE name and version:** CLion 2021.1 (CL-211.6432.9)
* **Operating system:** macOS 11.2.2
* **Macro expansion engine:** new
* **Name resolution engine:** new

Please let me know if such errors are not currently expected to be reported.
